### ISSUE TITLE
Add to cart tracking JS error (recobuy.js)

### ISSUE THREAD

**Issue by adamj88** [Nosto/nosto-magento2#294]
On the latest version of Magento OS (2.2.5) and latest version of the Nosto module (2.10.5).

Seeing an error occurring which stems from the following file and line: https://github.com/Nosto/nosto-magento2/blob/develop/view/frontend/web/js/recobuy.js#L40

```
var helper = new addToCart();
```

This is trying to call `new` on the following file which contains a jQuery widget, `mage.catalogAddToCart`: https://github.com/magento/magento2/blob/2.2.5/app/code/Magento/Catalog/view/frontend/web/js/catalog-add-to-cart.js

```
recobuy.js:40 Uncaught TypeError: addToCart is not a constructor
    at recobuy.js:40
    at Object.execCb (require.js:1650)
    at Module.check (require.js:866)
    at Module.<anonymous> (require.js:1113)
    at require.js:132
    at require.js:1156
    at each (require.js:57)
    at Module.emit (require.js:1155)
    at Module.check (require.js:917)
    at Module.enable (require.js:1143)
```

With merge or bundling enabled this causes the checkout to fail to load.

With merge and minify JS enabled:

```
Uncaught TypeError: addToCart is not a constructor
    at recobuy.min.js:1
    at Object.execCb (73a802a9d643b878b13427081a634902.min.js:112)
    at Module.check (73a802a9d643b878b13427081a634902.min.js:56)
    at Module.<anonymous> (73a802a9d643b878b13427081a634902.min.js:72)
    at 73a802a9d643b878b13427081a634902.min.js:11
    at 73a802a9d643b878b13427081a634902.min.js:74
    at each (73a802a9d643b878b13427081a634902.min.js:3)
    at Module.emit (73a802a9d643b878b13427081a634902.min.js:74)
    at Module.check (73a802a9d643b878b13427081a634902.min.js:61)
    at Module.enable (73a802a9d643b878b13427081a634902.min.js:73)
```

With bundle and minify JS enabled:

```
Uncaught TypeError: addToCart is not a constructor
    at eval (eval at require.load (static.min.js:9), <anonymous>:3:125)
    at Object.execCb (require.min.js:112)
    at Module.check (require.min.js:56)
    at Module.<anonymous> (require.min.js:72)
    at require.min.js:11
    at require.min.js:74
    at each (require.min.js:3)
    at Module.emit (require.min.js:74)
    at Module.check (require.min.js:61)
    at Module.enable (require.min.js:73)
```

**Comment by hannunosto** [comment]
Hi @adamj88, this usually happens when javascript merging & bundling is enabled and the static content has not been re-deployed after Nosto module installation. Can you please run `bin/magento setup:static-content:deploy` and let us know if that solves the issue.

**Comment by adamj88** [comment]
@hannunosto I've redeployed with static-content after module installation and the issue still persists.

Could you explain more about what the problem is and how redeploying static content might resolve this?

**Comment by hannunosto** [comment]
For some of our clients the static content re-deployment has solved similar issue(s). Anyways, there's clearly something off in initializing / constructing the widget. 

We have a fix on our way here: https://github.com/Nosto/nosto-magento2/pull/295

**Comment by hannunosto** [comment]
This has been fixed in version 2.10.6
